src/activities/ActivityUsageCollectionEntity.js

```diff
@@ -1,42 +1,63 @@
-'use strict';
-
-import { Entity } from '../es6/Entity.js';
-import { Rels } from '../hypermedia-constants';
-import { ActivityUsageEntity } from './ActivityUsageEntity.js';
-
-/**
- * ActivityUsageCollectionEntity class representation of a d2l activity usage collection.
- */
-export class ActivityUsageCollectionEntity extends Entity {
-	_items() {
-		if (!this._entity) {
-			return;
-		}
-		return this._entity.getSubEntitiesByRel('item');
-	}
-
-	onItemsChange(onChange) {
-		const items = this._items();
-
-		items.forEach((entity, index) => {
-			const onChangeWithIndex = collectedItem => {
-				onChange(collectedItem, index);
-			};
-			entity && this._subEntity(CollectedItemEntity, entity, onChangeWithIndex);
-		});
-	}
-}
-
-class CollectedItemEntity extends Entity {
-	activityUsageHref() {
-		if (!this._entity || !this._entity.hasLinkByRel(Rels.Activities.activityUsage)) {
-			return;
-		}
-		return this._entity.getLinkByRel(Rels.Activities.activityUsage).href;
-	}
-
-	onActivityUsageChange(onChange) {
-		const activityUsageHref = this.activityUsageHref();
-		activityUsageHref && this._subEntity(ActivityUsageEntity, activityUsageHref, onChange);
-	}
-}
+'use strict';
+
+import { Entity } from '../es6/Entity.js';
+import { Rels } from '../hypermedia-constants';
+import { ActivityUsageEntity } from './ActivityUsageEntity.js';
+import { performSirenAction } from '../es6/SirenAction.js';
+
+const actions = {
+	removeActivity: 'remove-activity'
+};
+
+/**
+ * ActivityUsageCollectionEntity class representation of a d2l activity usage collection.
+ */
+export class ActivityUsageCollectionEntity extends Entity {
+	_items() {
+		if (!this._entity) {
+			return;
+		}
+		return this._entity.getSubEntitiesByRel('item');
+	}
+
+	onItemsChange(onChange) {
+		const items = this._items();
+
+		items.forEach((entity, index) => {
+			const onChangeWithIndex = collectedItem => {
+				onChange(collectedItem, index);
+			};
+			entity && this._subEntity(CollectedItemEntity, entity, onChangeWithIndex);
+		});
+	}
+
+	removeItem(href) {
+		const newEntityList = [];
+		this._entity.entities.forEach((entity) => {
+			if (href === entity.getLinkByRel('self').href) {
+				const action = entity.getActionByName(actions.removeActivity);
+				performSirenAction(this._token, action).then((response) => {
+					this.update(response);
```

Comment:
This is to double check if the action was successful and update to element to make sure it matches the saved version

Reply:
Is this necessary? Shouldn't the response automatically trigger the `onChange` callback since it has the same `href`? Then we could call the action directly on the sub-entity rather than having to iterate through comparing `href`s.

Reply:
I think I should check if performSirenAction does something like this already. I sort of assumed it didn't.

Reply:
Oh, so I am doing this iteration so I can delete it from the entity at the same time. So the entity gets updated right away. Instead of waiting for a reply from the server since this action is going to pass 99% of the time we might as well make the experience super quick for those users.

Reply:
I could be mistaken but won't the promise not resolve until the new collection entity has been returned so calling update here won't save us any time, it'll just cause a double re-render?

Reply:
It won't re-render if nothing has changed. So we don't need to worry about that.

It does sent two out:
--> happens only on client side so everything is returned super fast and we don't need to wait.
--> Lets send out the action and wait for the response and call render if anything has changed.

Reply:
In the future it be nice to be able to popup an alert saying failed to delete and refresh the page back instead of just refreshing the page back.

Reply:
Off line talk. I miss understood and forgot about the first comment.